### ISSUE TITLE
under sampling

### ISSUE THREAD

**Issue by zambery** [mhwong2007/Deep-Packet#2]
Hi, I notice that in your code under sampling was made after "train and test split", and only done it with train set. Maybe this operation will make test set is bigger than train set. So this is a mistake, or there are some reasons that you did such operation?

**Comment by mhwong2007** [comment]
No, I did it on purpose. In machine learning, we assume that the data we had collected so far are generated by an unknown underlying distribution. To assess our model generalisation capability, one of the possible ways to split train and test set is to maintain their label distribution, in this case, stratified sampling.

Under sampling is just one of model training tricks, and should only be operated on training set.

Therefore, yes, the number of training data may be less than test data after under sampling.
